pkg/subscriptions/subscriptions.go

```diff
@@ -39,9 +38,18 @@ type subscriptionData struct {
 func (s subscriptionData) saveTo(dir string) error {
 	path := filepath.Join(dir, s.name)
 	if err := os.MkdirAll(filepath.Dir(path), s.dirMode); err != nil {
-		return err
+		return fmt.Errorf("create subscription directory: %w", err)
 	}
-	return os.WriteFile(path, s.data, s.mode)
+	if err := os.Chmod(dir, s.dirMode); err != nil {
+		return fmt.Errorf("chmod subscription directory: %w", err)
+	}
+	if err := os.WriteFile(path, s.data, s.mode); err != nil {
+		return fmt.Errorf("write subscription data: %w", err)
+	}
+	if err := os.Chmod(path, s.mode); err != nil {
```

Comment:
Is this necessary, since the WriteFile already use the s.mode?

Reply:
we need it, the `WriteFile` honors the umask (the s.mode is passed to `open`) which now can have a value different than 0